### DISCUSSION TITLE
Killclick only works half the time?

### DISCUSSION THREAD

**Discussion by o777l** [winstxnhdw/lc-hax#409] [General]
Just went into a room full of loot bugs and a thumper, enabled killclick and it did absolutely nothing when clicking

**Comment by winstxnhdw** [comment]
Maybe other players are holding ownership. Try latest, I am making sure to take ownership before killing.

**Reply by winstxnhdw** [reply]
Do note that if someone is also trying to steal ownership from you, this problem can still happen.

**Reply by o777l** [reply]
Oh ok. Btw was there a way that can add back the hitclick patch?

I have the previous versions with hitclick downloaded, but they only contain "'HitOnClickCommand.cs".

Can I just add it to the scripts/commands folder in the newest version or anything else needs changing?

**Reply by winstxnhdw** [reply]
I think you’ll need the previous `StunMod.cs`.

**Reply by o777l** [reply]
Tried but build is failing. Says "'Setting' does not contain a definition for 'EnableHitOnLeftClick'", and it's referring to lc-hax.csproj

**Reply by winstxnhdw** [reply]
You can add that field in `Setting.cs`.

**Reply by o777l** [reply]
Hm ok so I did it and compiled just fine, enables in game too. But it doesn't seem to do anything.

**Reply by winstxnhdw** [reply]
Did it used to work before?

**Reply by o777l** [reply]
I have a version from February 12 that had hitclick working. So now just trying to add it to the latest released version. I aslo added StunMod.cs

So this is the HitOnClickCommand.cs only, can't find other files referring to hitclick.

```
using Hax;

[Command ("hitclick")]
internal class HitOnClickCommand : ICommand {
    public void Execute(StringArray _) {
        Setting.EnableHitOnLeftClick = !Setting.EnableHitOnLeftClick;
        Chat.Print($"Hitclick: {(Setting.EnableHitOnLeftClick ? "Enabled" : "Disabled" )}");
```

**Reply by winstxnhdw** [reply]
The `hitclick` logic is mostly contained in `StunMod`. That file you are showing is just to toggle the flag that will allow you to use it. You need the old version of `StunMod`

**Reply by o777l** [reply]
I see, so here is what I have changed:

Added `HitOnClickCommand.cs` to commands folder
Updated `Setting.cs` with EnableHitOnLeftClick
Added old `StunMod.cs` to modules

**Reply by winstxnhdw** [reply]
Yup

**Reply by o777l** [reply]
Not working unfortunately

**Reply by winstxnhdw** [reply]
You have to do `/hitclick` first right?

**Reply by o777l** [reply]
Exactly, it says enabled but clicking enemies have no effect

**Reply by winstxnhdw** [reply]
Don't you have to have a defensive weapon equipped as well?

**Reply by o777l** [reply]
Using the old version from more than a week ago now, you can do it unequipped

**Reply by winstxnhdw** [reply]
Pretty sure you never could. The author who wrote that mod always intended it to only work with a defensive weapon equipped. If it worked before, means it was bugged somehow and somewhere along the way it got fixed.

**Reply by o777l** [reply]
Oh ok I mean I always used it that way, just spamming it to drop enemies. Never had to carry a shovel

**Reply by winstxnhdw** [reply]
I mean you can just delete the if-statement that checks for a defensive weapon.
